spark-plugins/src/test/java/co/cask/hydrator/plugin/spark/test/DecisionTreeRegressionTest.java

```diff
@@ -0,0 +1,338 @@
+      .addStage(new ETLStage("customsink", new ETLPlugin(DecisionTreeTrainer.PLUGIN_NAME, SparkSink.PLUGIN_TYPE,
+                                                         properties, null)))
+      .addConnection("source", "customsink")
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(DATAPIPELINE_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app("DecisionTreeTest");
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+
+    // send records from sample data to train the model
+    List<StructuredRecord> messagesToWrite = new ArrayList<>();
+    messagesToWrite.addAll(getInputData());
+
+    // write records to source
+    DataSetManager<Table> inputManager = getDataset(Id.Namespace.DEFAULT, "flightRecords");
+    MockSource.writeInput(inputManager, messagesToWrite);
+
+    // manually trigger the pipeline
+    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    workflowManager.start();
+    workflowManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    Assert.assertEquals("FAILED", workflowManager.getHistory().get(0).getStatus().name());
+  }
+
+  @Test
+  public void testIncludeAllFeatures() throws Exception {
+    /*
+     * source --> sparksink
+     */
+    Map<String, String> properties = new ImmutableMap.Builder<String, String>()
+      .put("fileSetName", "decision-tree-regression-model")
+      .put("path", "decisionTreeRegression")
+      .put("labelField", "delayed")
+      .put("maxBins", "100")
+      .put("maxDepth", "9")
+      .build();
+
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(new ETLStage("source", MockSource.getPlugin("flightRecords")))
+      .addStage(new ETLStage("customsink", new ETLPlugin(DecisionTreeTrainer.PLUGIN_NAME, SparkSink.PLUGIN_TYPE,
+                                                         properties, null)))
+      .addConnection("source", "customsink")
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(DATAPIPELINE_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app("SinglePhaseApp");
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+
+    // send records from sample data to train the model
+    List<StructuredRecord> messagesToWrite = new ArrayList<>();
+    messagesToWrite.addAll(getInputData());
+
+    // write records to source
+    DataSetManager<Table> inputManager = getDataset(Id.Namespace.DEFAULT, "flightRecords");
+    MockSource.writeInput(inputManager, messagesToWrite);
+
+    // manually trigger the pipeline
+    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    workflowManager.start();
+    workflowManager.waitForFinish(5, TimeUnit.MINUTES);
+    //featurs can only have type: int, long, float, double
+    Assert.assertEquals("FAILED", workflowManager.getHistory().get(0).getStatus().name());
```

Comment:
similar comment here. The config class should have its own test class for all the validation that needs to be done.